File: doacao_app_java/src/main/java/repository/MYSQLConnection.java
package repository;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class MYSQLConnection {
    private static Connection connection;

    private MYSQLConnection() {}

    public static Connection getConnection() {
        if (connection == null) {
            try {
                String url = "jdbc:mysql://localhost:3306/doacao_animal";
                String user = "root";
                String password = "123456";
                connection = DriverManager.getConnection(url, user, password);
                System.out.println("Conexão realizada com sucesso!");
            } catch (SQLException e) {
                e.printStackTrace();
            }
        }
        return connection;
    }

    public static void closeConnection() {
        try {
            if (connection != null) {
                connection.close();
                connection = null;
            }
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }

    public static void main(String[]args){
        getConnection();
        closeConnection();
    }
}

File: doacao_animal_python/repository/mysql_connection.py
import mysql.connector
from mysql.connector import Error

class MYSQLConnection():
    _connection = None
    
    @staticmethod
    def get_connection():
        if MYSQLConnection._connection is None:
            try:
                MYSQLConnection._connection = mysql.connector.connect(
                    host= "localhost",
                    database= "doacao_animal",
                    user= "root",
                    password="123456"
                )
                print("Conexão bem-sucedida ao banco de dados MySQL")
            except Error as e:
                print(f"Erro ao conectar ao MySQL: {e}")
        return MYSQLConnection._connection

    @staticmethod
    def close_connection():
        if MYSQLConnection._connection is not None and MYSQLConnection._connection.is_connected():
            MYSQLConnection._connection.close()
            MYSQLConnection._connection = None
            print("Conexão ao banco de dados MySQL fechada")
    